packages/gatsby-plugin-gatsby-cloud/src/indicator.js

```diff
@@ -0,0 +1,342 @@
+      <path
+        d="M14.6663 6.2735L13.7263 5.3335L9.99967 9.06016L6.27301 5.3335L5.33301 6.2735L9.05967 10.0002L5.33301 13.7268L6.27301 14.6668L9.99967 10.9402L13.7263 14.6668L14.6663 13.7268L10.9397 10.0002L14.6663 6.2735Z"
+        fill="white"
+      />
+    </svg>
+  </>
+)
+
+export default class Indicator extends React.Component {
+  state = {
+    attributes: {},
+    icon: ``,
+  }
+
+  componentDidMount() {
+    const prettyUrlRegex = /^preview-/
+    const host = window.location.hostname
+    let buildId
+
+    try {
+      setInterval(
+        async () => {
+          // currentBuild is the most recent build that is not QUEUED
+          // latestBuild is the most recent build that finished running (ONLY status ERROR or SUCCESS)
+          const { currentBuild, latestBuild } = await getBuildInfo()
+
+          if (!buildId) {
+            if (prettyUrlRegex.test(host)) {
+              buildId = latestBuild?.id
+            } else {
+              const buildIdMatch = host.match(/build-(.*?(?=\.))/)
+              buildId = buildIdMatch && buildIdMatch[1]
+            }
+          }
+
+          if (currentBuild?.buildStatus === `BUILDING`) {
+            this.setState(prevState =>
+              Object.assign({}, prevState, {
+                attributes: indicatorSetBuilding(),
+                icon: <Spinner />,
+              })
+            )
+          } else if (currentBuild?.buildStatus === `ERROR`) {
+            this.setState(prevState =>
+              Object.assign({}, prevState, {
+                attributes: indicatorSetFailed(),
+                icon: <FailedIcon />,
+              })
+            )
+          } else if (buildId === currentBuild?.id) {
+            this.setState(prevState =>
+              Object.assign({}, prevState, {
+                attributes: indicatorSetUpToDate(),
+                icon: ``,
+              })
+            )
+          } else if (
+            buildId !== latestBuild?.id &&
+            latestBuild?.buildStatus === `SUCCESS`
+          ) {
+            this.setState(prevState =>
+              Object.assign({}, prevState, {
+                attributes: indicatorSetSuccess(latestBuild?.id),
+                icon: <SuccessIcon />,
+              })
+            )
+          }
+        },
+        process.env.NODE_ENV === `test` ? 10 : POLLING_INTERVAL
+      )
+    } catch (e) {
+      console.log(e)
+    }
+  }
+
+  render() {
+    if (Object.keys(this.state.attributes).length === 0) {
+      return <>{this.props.children}</>
+    } else {
+      return (
+        <>
+          <Style />
+          <div
+            data-testid="preview-status-indicator"
+            onClick={this.state.attributes.onclick}
+            style={{
+              color: this.state.attributes.color,
+              backgroundColor: this.state.attributes.backgroundColor,
+              cursor: this.state.attributes.cursor,
+            }}
+            data-gatsby-loading-indicator="root"
```

Comment:
You'll want to change these data attributes to `data-gatsby-preview-indicator` (+ change the styles). Otherwise we'd have the same selectors twice -- shouldn't hurt as styles are scoped to the portal but better safe than sorry